### PR TITLE
Update media extractors

### LINKED ISSUE #940: Loading PDF. RAM consumption.
When I upload a large PDF file (for example a book with 300 pages), memory usage become incredible.
It takes all available RAM (about 16 gigabytes), starts to use swap and my system literally die. 

Probably we should not store all pages in memory during extraction

### PR DESCRIPTION
**Pull request by zhiltsov-max**
### Motivation and context

Follow up for #2141 

- Added PDF import tests
- Fixed #940 
- Fixed a problem with pdf cached chunks - after the first interpreter exit, extracted files were removed, making it impossible to recreate chunks
- Refactored some code

zip - need to test if extracting is really necessary

### How has this been tested?

Unit tests, manual tests

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [x] I have added tests to cover my changes
- [x] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2020 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by nmanovic** [comment]
@zhiltsov-max , could you please help to fix merge conflicts?

**Comment by nmanovic** [comment]
@Marishka17 , could you please review and test the PR?

File: cvat/apps/engine/media_extractors.py [hunk 1]
```diff
@@ -65,9 +66,16 @@ def _get_preview(obj):
         return preview.convert('RGB')
 
     @abstractmethod
-    def get_image_size(self):
+    def get_image_size(self, i):
         pass
 
+    def __len__(self):
+        return (self._stop - self._start) // self._step
```
Comment (Marishka17) [inline cvat/apps/engine/media_extractors.py:73]: I think that it is not equal  `len(frame_range)`, but I could be wrong.

**Review by Marishka17 — APPROVED** [review]

**Comment by bsekachev** [comment]
@zhiltsov-max @Marishka17 @azhavoro What is status of this PR? Can me merge it?

**Review by azhavoro — APPROVED** [review]

**Pull request merged**